sktime/clustering/partitioning/_time_series_k_partition.py

```
@@ -0,0 +1,250 @@
+                then must be of type Init_Algo
+
+            max_iter: int, default = 300
+                Maximum number of iterations of time series k means
+                for a single run.
+
+            verbose: bool, default = False
+                Verbosity mode.
+
+            metric: Metric_Parameter, default = None
+                The distance metric that is used to calculate the
+                distance between points.
+
+        """
+        super(TimeSeriesKPartition, self).__init__()
+
+        self.n_clusters = n_clusters
+        self.max_iter = max_iter
+        self.verbose = verbose
+        self.metric = metric
+        self.init_algorithm = init_algorithm
+
+        self.__centers: Numpy_Array = None
+        self._init_algorithm = None
+        self._metric = None
+
+    def _fit(self, X: Numpy_Array) -> None:
+        """
+        Method that is used to fit the time series k
+        partition model on dataset X
+
+        Parameters
+        ----------
+        X: Numpy_Array
+            Numpy array of series to train the model on
+
+        Returns
+        -------
+        self
+            Fitted estimator
+        """
+        center_algo: BaseClusterCenterInitializer = self._init_algorithm(
+            X, self.n_clusters
+        )
+        self.__centers = center_algo.initialize_centers()
+
+        for _ in range(self.max_iter):
+            self.__update_centers(X)
+
+    def _predict(self, X: Numpy_Array) -> Numpy_Array:
+        """
+        Method used to perform a prediction from the trained
+        time series k partition clustering algorithm
+
+        Parameters
+        ----------
+        X: Numpy_Array
+            Numpy_Array containing the time series to
+            predict clusters for
+
+        Returns
+        -------
+        Numpy_Array
+            Index of the cluster each sample belongs to
+        """
+
+        if self.__centers is None:
```

This should be handled via `self.check_is_fitted()`. If you still want to raise a separate exception here, please make it more specific, e.g. `AttributeError`